src/server/models.py

```diff
@@ -2,12 +2,116 @@
+        Type of pattern to use for file filtering (include or exclude).
+    pattern : str
+        Glob/regex pattern string for file filtering.
+    token : str | None
+        GitHub personal access token (PAT) for accessing private repositories.
+
+    """
+
+    input_text: str = Field(..., description="Git repository URL or slug to ingest")
+    max_file_size: int = Field(..., ge=0, le=500, description="File size slider position (0-500)")
+    pattern_type: PatternType = Field(default=PatternType.EXCLUDE, description="Pattern type for file filtering")
+    pattern: str = Field(default="", description="Glob/regex pattern for file filtering")
+    token: str | None = Field(default=None, description="GitHub PAT for private repositories")
+
+    @field_validator("input_text")
+    @classmethod
+    def validate_input_text(cls, v: str) -> str:
+        """Validate that input_text is not empty."""
+        if not v.strip():
+            err = "input_text cannot be empty"
+            raise ValueError(err)
+        return v.strip()
+
+    @field_validator("pattern")
+    @classmethod
+    def validate_pattern(cls, v: str) -> str:
+        """Validate pattern field."""
+        return v.strip() if v else ""
+
+
+class IngestSuccessResponse(BaseModel):
+    """Success response model for the /api/ingest endpoint.
+
+    Attributes
+    ----------
+    repo_url : str
+        The original repository URL that was processed.
+    short_repo_url : str
+        Short form of repository URL (user/repo).
+    summary : str
+        Summary of the ingestion process including token estimates.
+    tree : str
+        File tree structure of the repository.
+    content : str
+        Processed content from the repository files.
+    default_file_size : int
+        The file size slider position used.
+    pattern_type : str
+        The pattern type used for filtering.
+    pattern : str
+        The pattern used for filtering.
+
+    """
+
+    repo_url: str = Field(..., description="Original repository URL")
+    short_repo_url: str = Field(..., description="Short repository URL (user/repo)")
+    summary: str = Field(..., description="Ingestion summary with token estimates")
+    tree: str = Field(..., description="File tree structure")
+    content: str = Field(..., description="Processed file content")
+    default_file_size: int = Field(..., description="File size slider position used")
+    pattern_type: str = Field(..., description="Pattern type used")
+    pattern: str = Field(..., description="Pattern used")
+
+
+class IngestErrorResponse(BaseModel):
+    """Error response model for the /api/ingest endpoint.
+
+    Attributes
+    ----------
+    error : str
+        Error message describing what went wrong.
+    repo_url : str
+        The repository URL that failed to process.
+
+    """
+
+    error: str = Field(..., description="Error message")
+    repo_url: str = Field(..., description="Repository URL that failed")
```

Comment:
I think ruff might want

```suggestion
    error: str = Annotated[str, Field(..., description="Error message")]
    repo_url: str = Annotated[str, Field(..., description="Repository URL that failed")]
```

with

```python
from gitingest.utils.compat_typing import Annotated
```